crates/rattler_conda_types/src/match_spec/parse.rs

```diff
@@ -380,22 +405,25 @@ fn matchspec_parser(
     let (input, _comment) = strip_comment(input);
     let (input, _if_clause) = strip_if(input);
 
-    // 2. Is the spec a tarball?
+    // 2. Is the spec a package file, then we can parse it as a url and use as is
     if is_package_file(input) {
-        let _url = match Url::parse(input) {
-            Ok(url) => url,
-            #[cfg(target_arch = "wasm32")]
-            Err(_) => return Err(ParseMatchSpecError::InvalidPackagePathOrUrl),
-            #[cfg(not(target_arch = "wasm32"))]
-            Err(_) => match PathBuf::from_str(input) {
-                Ok(path) => Url::from_file_path(path)
-                    .map_err(|_err| ParseMatchSpecError::InvalidPackagePathOrUrl)?,
-                Err(_) => return Err(ParseMatchSpecError::InvalidPackagePathOrUrl),
-            },
-        };
+        let url = PathBuf::from_str(input)
```

Comment:
You can't use `PathBuf::from_str` because it is platform-dependent. E.g. `C:\Windows\bla` is parsed differently on Unix and Windows.

Reply:
Fixed, using file_url now